### PR TITLE
[stable24] Properly calculate primary element based on background luminance

### PR DESCRIPTION
**Pull request by skjnldsv**
| Before | After |
|:---------:|:------:|
|Capture d’écran_2022-05-20_09-09-55|Capture d’écran_2022-05-20_09-05-47|

### CODE REVIEW

File: apps/accessibility/lib/Controller/AccessibilityController.php [hunk 1]
```diff
@@ -238,4 +238,15 @@ private function getInjectedVariables(): string {
...
 	}
+
+	/**
+	 * Return true if the dark theme is enabled for the current user
+	 */
+	private function isDarkThemeEnabled(): bool {
+		if (!$this->userSession->isLoggedIn()) {
+			return false;
+		}
+		$user = $this->userSession->getUser();
```
Comment (CarlSchwan) [inline apps/accessibility/lib/Controller/AccessibilityController.php:249]:
```suggestion
		$user = $this->userSession->getUser();
		if (!$user) {
			return false;
		}
```

File: apps/theming/lib/ThemingDefaults.php [hunk 2]
```diff
@@ -315,7 +315,7 @@ public function getFDroidClientUrl() {
 	/**
 	 * @return array scss variables to overwrite
 	 */
-	public function getScssVariables() {
+	public function getScssVariables($brightBackground = true) {
```
Comment (CarlSchwan) [inline apps/theming/lib/ThemingDefaults.php:318]:
```suggestion
	public function getScssVariables(bool $brightBackground = true) {
```

File: lib/private/legacy/OC_Defaults.php [hunk 3]
```diff
@@ -311,9 +311,9 @@ public function getColorPrimary() {
 	/**
 	 * @return array scss variables to overwrite
 	 */
-	public function getScssVariables() {
+	public function getScssVariables($brightBackground = true) {
```
Comment (CarlSchwan) [inline lib/private/legacy/OC_Defaults.php:314]:
```suggestion
	public function getScssVariables(bool $brightBackground = true) {
```

File: themes/example/defaults.php [hunk 4]
```diff
@@ -119,7 +119,7 @@ public function getColorPrimary() {
 	 * Returns variables to overload defaults from core/css/variables.scss
 	 * @return array
 	 */
-	public function getScssVariables() {
+	public function getScssVariables($brightBackground) {
```
Comment (CarlSchwan) [inline themes/example/defaults.php:122]:
```suggestion
	public function getScssVariables(bool $brightBackground) {
```

**Review by CarlSchwan — APPROVED** [review]

**Comment by skjnldsv** [comment]
Ping :)

**Review by artonge — APPROVED** [review]

**Pull request merged**